### ISSUE TITLE
Sticky keys without lock

### ISSUE THREAD

**Issue by CSRaghunandan** [pqrs-org/Karabiner-Elements#477]
I really miss this feature from karabiner. When will this feature come to karabiner elements? 
Also, is there any other way for me to use sticky keys without lock in macOS Sierra

**Comment by stifynsemons** [comment]
+1

Sticky keys help alleviate hand strain, and I used them heavily in Karabiner.

**Comment by snarfed** [comment]
i'm the one who added this to the original karabiner in tekezo/Karabiner#118, via its exposed sticky keys virtual key codes, e.g. `VK_STICKY_COMMAND_L`. [here's the full list.](https://github.com/tekezo/Karabiner/blob/master/src/bridge/generator/keycode/data/KeyCode.data#L314-L349)

@tekezo mentioned in https://github.com/tekezo/Karabiner-Elements/issues/551#issuecomment-272185300 that elements already handles the virtual keyboard so that it can explicitly support mac os x's sticky modifier keys. given that, we may just need `VK_STICKY_*` to be exposed in [`simple_modifications.json`](https://github.com/tekezo/Karabiner-Elements/blob/master/src/apps/PreferencesWindow/PreferencesWindow/Resources/simple_modifications.json)? not sure.

this is the only thing keeping me from switching to elements, btw. it's come a long way since i looked at it. congrats on the stable release, all!

**Comment by paulistoan** [comment]
+1, would love to have this feature

**Comment by jubilatious1** [comment]
+1, would love to have this feature also.

**Comment by vschumaker** [comment]
+1

**Comment by snarfed** [comment]
still valid, please keep.

**Comment by CSRaghunandan** [comment]
I still hope somebody takes over this task
ssshh bot!

**Comment by stembrain** [comment]
ive been googling for days to find out if KE can do sticky modifier keys. it really would help with wrist strain. please keep this alive :)

**Comment by paulistoan** [comment]
You might want to try a hardware approach such as a TMK usb to usb keyboard
adapter: https://geekhack.org/index.php?topic=69169.0 I haven't tried this
myself but it looks like it might do what you want.


Paul

**Comment by stembrain** [comment]
I see that OS X has its own built in sticky modifier keys. I’m going to mess with those, they do exactly what I wanted. Thank you Paul!

**Comment by snarfed** [comment]
@stembrain true, but OS X's built in feature also has a "lock" feature that i at least often enable unintentionally. more background in https://apple.stackexchange.com/questions/13708/disabling-the-keep-pressed-function-of-os-x-sticky-keys . this issue is to port the original karabiner sticky keys feature that *didn't* have that locking.

**Comment by stembrain** [comment]
Ah right I see that now that I used OS X sticky keys for a day. The way it has a one-shot (as ergodox ez keyboard document calls it) and a lock mode is a bit of a bother.

**Comment by stembrain** [comment]
The other weirdness with the OS X one is that shift becomes locked differently than the other modifiers. It takes one less press!

**Comment by jubilatious1** [comment]
Still hoping in Karabiner Elements someone can implement sticky modifier keys, on a per key basis. 
(OSX only does it keyboard-wide, not key-by-key).

**Comment by CSRaghunandan** [comment]
Please keep. Still hoping someone can work on this :)

**Comment by jubilatious1** [comment]
shhh, bot!

**Comment by jubilatious1** [comment]
Please keep active.

**Comment by jubilatious1** [comment]
Please keep active.

**Comment by jubilatious1** [comment]
Please keep active.

**Comment by knoajp** [comment]
I'm still keeping original Karabiner.
I'm very thirsty for this feature. Or are there anyone has requested for Apple to support "without lock"?

**Comment by shartf** [comment]
Apple has other problems with sticky keys - their unlock delay is so bad, you get 2-3 capital letters from one shift key with sticky keys with a reasonable typing speed. I also hope that this feature will be implemented, since it is very comfortable, if done properly.

**Comment by jubilatious1** [comment]
Please keep active.

**Comment by mgoszcz2** [comment]
I would still absolutely love this feature

**Comment by mgoszcz2** [comment]
Shush!

**Comment by stephanoparaskeva** [comment]
Any updates?

**Comment by knoajp** [comment]
I wonder this feature is actually possible on Karabiner-Elements or not.
If possible, why not!! (yes it needs developing resource...)

**Comment by knoajp** [comment]
Today I called Apple Japan to request to support this feature on macOS officially.

This is rather minor preference and the lock feature takes time for explaining to a staff who doesn't know it. At the first, the staff also confused, but she has comprehended on my second explaining.

She said the only thing we can is to send feedbacks to Apple.

Folks, let's send our earnest requests to Apple!!

📞 Telephone:
https://support.apple.com/HT201232
📝 Text Form:
https://www.apple.com/feedback/macos.html

**Comment by knoajp** [comment]
I also tweeted to Apple and @tekezo.

ja:
https://twitter.com/kantankikaku/status/1311899466870390786
en:
https://twitter.com/kantankikaku/status/1311900693679140871

**Comment by tekezo** [comment]
I added this feature to my tasks.
https://github.com/pqrs-org/Karabiner-Elements/blob/master/TASKS.md

**Comment by mgoszcz2** [comment]
I opened a radar about it in January (FB7548753)

**Comment by knoajp** [comment]
I had kept my Mac with El Capitan for a long time to use the predecessor app Karabiner, but jumped to Mojave the day before yesterday.

So I didn't understand what @shartf said about, but now I did.
https://github.com/pqrs-org/Karabiner-Elements/issues/477#issuecomment-541582317

It doesn't occur on El Capitan even if I switch on the official Sticky Keys preference of OS X. And I found by google that it has begun on Sierra.

Then I again called Apple Japan to report this bug today. May it be fixed as soon as possible... or Karabiner-Elements support it. ;-)

**Comment by nkgm** [comment]
I'm aware the original title says "without lock" but I really miss the good old `VK_LOCK_MODIFIER_FORCE_ON/OFF` from Karabiner Classic, which I constantly find myself working around in KE. Seeing that @tekezo will be spending time on this feature in the near future, maybe it's a good time to squeeze in lock functionality as well?

**Comment by tekezo** [comment]
Sticky modifiers has been supported at v13.3.0!
https://karabiner-elements.pqrs.org/docs/help/how-to/sticky-modifier-key/

**Comment by jubilatious1** [comment]
Thank you @tekezo !!

**Comment by knoajp** [comment]
It works well on keyboard keys.
But its behavior on mouse clicks is different from it of macOS or Windows default.

1. Press [shift] key (sticky [shift] is on)
2. Click somewhere (the click event is fired with shift key)
3. Sticky [shift] becomes off

Current Karabiner-Elements lacks the 3. feature.
(it needs another [shift] key pressed to cancel the sticky status)

**Comment by quarkw** [comment]
> It works well on keyboard keys. 👍
> But its behavior on mouse clicks is different from it of macOS or Windows default.
> 
>     1. Press [shift] key (sticky [shift] is on)
> 
>     2. Click somewhere (the click event is fired with shift key)
> 
>     3. Sticky [shift] becomes off
> 
> 
> Current Karabiner-Elements lacks the 3. feature.
> (it needs another [shift] key pressed to cancel the sticky status)

This is actually supported by Karabiner-Elements for external mouses:
https://karabiner-elements.pqrs.org/docs/help/how-to/sticky-modifier-key/#sticky-modifiers-and-mouse-events

There is no way to enable the mouse device on the internal trackpad, so it seems like there needs to be an alternative solution for the trackpad

**Comment by ihventura** [comment]
Please keep this one alive. We really need sticky keys WITHOUT keylock! C'mon, I'm dying over here!
